2.0.0: Move away from @DbRef relations

demoth opened the pull request:
Fix /invite to space

Update all libs

Fix Validation Error handling

demoth committed:
1.5.1: Add users to space with /space/invide url

demoth committed:
move user logic into `UserService`

demoth committed:
1.5.2 Spaces

manage spaces by `/space` endpoint

demoth committed:
1.6.0: Migrate from `@DbRef` mongo relations

+ replace all db refs with pojo data classes to avoid cycles

+ migrate all existing data to new db model version

denolia commented on build.gradle.kts:
```diff
@@ -11,7 +11,7 @@ plugins {
 
 group = "org.dnj"
-version = "1.5.0"
+version = "1.5.1"
```
я протестую! для такого глобального изменения нужно бампнуть хотя бы минорную цифру, а лучше - мажорную

demoth replied:
Хорошо :)

denolia commented on src/main/kotlin/org/dnj/memoria/Initializer.kt:
```diff
@@ -19,21 +22,72 @@ class Initializer(
+            space.owner?.let { owner ->
+                space.ownerRef = owner.toDto()
+                space.owner = null
+            }
```
Can we do the same for `space.participants`?

denolia commented on src/main/kotlin/org/dnj/memoria/controllers/SpaceController.kt:
```diff
@@ -95,13 +95,13 @@ class SpaceController(
 
         // anyone already in the space can invite
-        if (!owner.spaces.contains(space))
+        if (owner.spaceRefs.none { s -> s.id == space.id })
```
It looks like the comment and the code do not correspond to each other

denolia commented on src/main/kotlin/org/dnj/memoria/model/space.kt:
```diff
@@ -12,11 +12,21 @@ data class Space(
+
+    var version: String? = "1",
+
+    var ownerRef: UserDto? = null
```
let's add participantRefs  :)

demoth committed:
update all libraries

demoth committed:
remove redundant opt-in annotations (after kotlin update)

demoth committed:
forgot to migrate item.space

demoth committed:
cleanup unused repository methods

demoth committed:
+ fix item repository find by space
+ fix validation error handling when updating the item
+ fix migration issuen when applied 2 times

demoth committed:
+ fix item repository find by space
+ fix validation error handling when updating the item
+ fix migration issuen when applied 2 times

demoth committed:
+fix space ref nullability when receiving from web

demoth committed:
authenticate users when get all users
